source/gui/addonGui.py

```diff
@@ -206,18 +221,26 @@ def refreshAddonsList(self,activeIndex=0):
 			self.helpButton.Disable()
 			self.removeButton.Disable()
 
+	def _shouldDisable(self, addon):
+		return not (addon.isPendingDisable or (addon.isDisabled and not addon.isPendingEnable and not addon.isPendingDisable))

```

Comment:
This is a confusing bit of logic. Do you need to check addon.isPendingDisable twice? I would have thought that if it got to the inner parenthesis, isPendingDisable could only be false?